### ISSUE TITLE
The order of adding permissions influences the checks

### ISSUE THREAD

**Issue by mfarbarn** [entrecode/shiro-trie#147]
Hi,

Depending on the order of adding permissions to the tree, the tree may not be correctly constructed. As a result, some checks may fail while they should succeed.

```
This works as expected:

var shiroTrie = require('shiro-trie');

var account1 = shiroTrie.newTrie();

account1.add([
  'domain:*',
  '*:*:read',
]);

account1.check('domain:resource:create'); // true
```

But this fails:
```
var shiroTrie = require('shiro-trie');

var account1 = shiroTrie.newTrie();

account1.add([
  '*:*:read',
  'domain:*', // domain does not get a * leaf
]); 

account1.check('domain:resource:create'); // false
```

**Comment by deyhle** [comment]
Thank you for reporting this bug. I'll look into it if I find the time. Feel free to create a Pull Request if you already have a fix for this.

**Comment by loxy** [comment]
Is this project dead?

**Comment by deyhle** [comment]
I just released v0.4.10 to npm which should fix this issue. Sorry for the delay! Feel free to check if that solves your issue @mfarbarn
